kong/plugins/aws-lambda/schema.lua

```diff
@@ -95,6 +95,12 @@ return {
           type = "boolean",
           default = true,
         } },
+        { aws_imds_protocol_version = {
+          type = "string",
+          required = true,
+          default = "v1",
+          one_of = { "v1", "v2" }
+        } },
```

Comment:
Why is the code asking for this input from the user?
Why is the code not smart enough to use v2 first and then fallback on v1 if v2 is not available?
I don't see any difference in functionality (so far as the current use of the API is concerned) between the two versions.

Am I missing something?

Reply:
By default, EC2 instances support both v1 and v2, and it is a concious decision for the AWS administrator to disable v1.  While it would generally be possible to default to v2, this would create the problem that if Kong is running inside a container on the EC2 instance, the IMDS request will hang and eventually time out.  I think this would be difficult to debug, and as the switch to v2-only is a configuration option that needs to be conciously made, requiring the plugin to be configured accordingly seems sensible.

Reply:
> this would create the problem that if Kong is running inside a container on the EC2 instance, the IMDS request will hang and eventually time out.

What problem do we run into if v2 request is made with a reasonable timeout and if the timeout is reached then v1 is used for the lifetime of the process?

Reply:
> What problem do we run into if v2 request is made with a reasonable timeout and if the timeout is reached then v1 is used for the lifetime of the process?

The first request to the plugin will be delayed by the timeout.

Reply:
Yes and that sounds like an acceptable compromise given that we make the user care about one less thing.
What do you think?

Reply:
I'm not convinced as the behavior would be kind of magical and our supporting code would be more complex.  That said, if you want me to make that change, I'll look into it next week.

Reply:
After a conversation on Slack, we've decided to leave things as they are - with explicit configuration - as automatically detecting the IMDS protocol version would potentially impact Kong's startup time too much.